### PR TITLE
Support for Swift Package Manager!

### LINKED ISSUE #82: Support for Swift Package Manager?
I want to manage BartyCrouch with [Mint](https://github.com/yonaskolb/Mint).
It needs SPM support for these commands. 
But BartyCrouch doesn't support it.

Does BartyCrouch have a plan to support SPM?

### PR DESCRIPTION
**Pull request by mshibanami**
Resolves #82 

## TL;DR

I created Package.swift for SPM. 
And to do it, I modified some source files.
But I didn't add the test target `BartyCrouchTests` for SPM because of a ObjC file. 

---

First of all, I wrote `target` in Package.swift like this:

```swift
        .target(
            name: "BartyCrouchKit",
            dependencies: [
            ],
            path: ".",
            sources: [
                "Sources/Code",
                "Carthage/CommandLine",
                "Carthage/Handyswift",
                "Carthage/Polyglot"
            ]
        ),
        .target(
            name: "BartyCrouch CLI",
            dependencies: [
                "BartyCrouchKit"
            ],
            path: ".",
            sources: [
                "Sources/Code",
                "Carthage/CommandLine",
                "Carthage/Handyswift",
                "Carthage/Polyglot",
                "BartyCrouch CLI/"
                ]
            ),
```

This is almost the same as the Xcode's setting `Compile Sources`, 
But it shows the following error.

```swift
error: target 'BartyCrouch CLI' has sources overlapping sources: /Users/abc/workspace/BartyCrouch/Carthage/Handyswift/Extensions/CharacterViewExtension.swift, ...
```

It looks SPM doesn't allow sharing same source files for multiple targets.
So I modified the code like this:

```
.target(
    name: "BartyCrouchKit",
    dependencies: [
    ],
    path: ".",
    sources: [
        "Sources/Code",
        "Carthage/CommandLine",
        "Carthage/Handyswift",
        "Carthage/Polyglot"
    ]
),
.target(
    name: "BartyCrouch CLI",
    dependencies: [
        "BartyCrouchKit"
    ],
    path: ".",
    sources: [
        "BartyCrouch CLI/"
        ]
    ),
```

Then `BartyCrouch CLI/main.swift` needed  `import BartyCrouchKit`. 
And CommandLineActor needed a public initializer.
So I added them.

I made sure this PR works fine in my environment. Building on Xcode and CocoaPods as well.

But I didn't make `testTarget`
I tried to write a `target` for tests like this:

```swift
.testTarget(
    name: "BartyCrouchKitTests",
    dependencies: [
        "BartyCrouchKit"
    ],
    path: "Tests"
)
```

But when I ran `swift test` then this error was displayed:

> error: target at '/Users/abc/workspace/BartyCrouch/Tests' contains mixed language source files; feature not supported

It looks SPM doesn't allow ObjC files, i.e. `Tests/Supporting Files/Constants.m`.

In the file, a preprocessor macro `PROJECT_DIR` is used.
It is why it is not a swift file, right? You know swift can't handle it.

Probably there are some nicer ways to get rid of the macro. But I think it makes pretty big refactoring.
And unit tests on Xcode is still available.
So I didn't add `testTarget` to Package.swift.
Sorry for my laziness...

### CODE REVIEW

**Comment by Dschee** [comment]
Thank you for the PR! I will need to check that the changes still work via Homebrew (especially I worry about the `import BartyCrouchKit` in main.swift) but if that works, I'm totally fine with a working not-so-perfect solution to the problem for now, which doesn't include tests and doesn' specify the dependencies (like HandySwift) as dependencies. We can improve later. Great work so far!
